promql/engine.go

```diff
@@ -720,6 +745,14 @@ func (ev *evaluator) rangeEval(f func([]Value, *EvalNodeHelper) Vector, exprs ..
 		enh.ts = ts
 		result := f(args, enh)
 		enh.out = result[:0] // Reuse result vector.
+
+		// In some cases we only have the result of 'f' without having tracked any samples
+		// in the above for loop that built the input vectors.
+		ev.currentSamples += len(result)
+		if ev.currentSamples > ev.maxSamples {
+			ev.error(error(ErrTooManySamples(env)))
```

Comment:
Why do you need the `error()` here?

Reply:
do we not have more samples in memory via the return value of `f`?

Reply:
I mean specifically, why are you wrapping `ErrTooManySamples` in `error`?

Reply:
Hmm, at some point I had a compile error with this but you're right they're not needed.